grad_check.py

```diff
@@ -13,28 +13,41 @@
 parser.add_argument('-c', '--cuda', action='store_true')
 options = parser.parse_args()
 
+cpp_module_path = os.path.dirname(
+    pkg_resources.resource_filename(
+        pkg_resources.Requirement.parse('lltm_cpp'), "lltm_cpp.py"))
+cpp_lib_path = glob.glob(os.path.join(cpp_module_path, "lltm_cpp*.so"))[0]
+torch.ops.load_library(cpp_lib_path)
+
+cuda_module_path = os.path.dirname(
+    pkg_resources.resource_filename(
+        pkg_resources.Requirement.parse('lltm_cuda'), "lltm_cuda.py"))
+cuda_lib_path = glob.glob(os.path.join(cuda_module_path, "lltm_cuda*.so"))[0]
+torch.ops.load_library(cuda_lib_path)
+
+
 if options.example == 'py':
     from python.lltm_baseline import LLTMFunction
-elif options.example == 'cpp':
-    from cpp.lltm import LLTMFunction
+    lltm_func = LLTMFunction.apply
 else:
-    from cuda.lltm import LLTMFunction
-    options.cuda = True
+    lltm_func = torch.ops.myops.lltm
+
+options.cuda |= (options.example == "cuda")
 
 device = torch.device("cuda") if options.cuda else torch.device("cpu")
 
 kwargs = {'dtype': torch.float64,
           'device': device,
           'requires_grad': True}
 
-X = torch.randn(options.batch_size, options.features, **kwargs)
-h = torch.randn(options.batch_size, options.state_size, **kwargs)
-C = torch.randn(options.batch_size, options.state_size, **kwargs)
-W = torch.randn(3 * options.state_size, options.features + options.state_size, **kwargs)
-b = torch.randn(1, 3 * options.state_size, **kwargs)
+X = torch.randn(options.batch_size, options.features, **kwargs).to(device)
```

Comment:
is the `.to(evice)` needed here, given that the device option is already present in the kwargs ?